provider/vsphere/provider.go

```diff
@@ -67,11 +68,45 @@ func (p environProvider) CloudSchema() *jsonschema.Schema {
 	return cloudSchema
 }
 
+const failedLoginMsg = "ServerFaultCode: Cannot complete login due to an incorrect user name or password."
+
 // Ping tests the connection to the cloud, to verify the endpoint is valid.
 func (p environProvider) Ping(endpoint string) error {
-	// a stub for now until we get a real implementation.
-	_, err := url.Parse(endpoint)
-	return errors.Trace(err)
+	// try to be smart and not punish people for adding or forgetting http
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return errors.New("Invalid endpoint format, please give a full url or IP/hostname.")
```

Comment:
Yes, but it interprets the string as the URL path.

"Parse parses rawurl into a URL structure. The rawurl may be relative or absolute."